bindings/java/src/org/sleuthkit/datamodel/blackboardutils/attributes/GeoTrackPointList.java

```diff
@@ -0,0 +1,293 @@
+	 *
+	 * @param jsonString JSon string of track points.
+	 *
+	 * @return	Timestamp ordered list of GeoTrackPoints, empty list will be
+	 *         returned if jsonString is null or empty.
+	 */
+	public static GeoTrackPointList deserialize(String jsonString) {
+		if (jsonString == null || jsonString.isEmpty()) {
+			return null;
+		}
+
+		return (new Gson()).fromJson(jsonString, GeoTrackPointList.class);
+	}
+	
+	/**
+	 * Construct an empty GeoTrackPointList.
+	 */
+	public GeoTrackPointList() {
+		points = new ArrayList<>();
+	}
+	
+	/**
+	 * Construct a new instance with the given list of GeoTrackPoint objects.
+	 * 
+	 * @param points List of track points, cannot be null.
+	 */
+	public GeoTrackPointList(List<GeoTrackPoint> points) {
+		if(points == null) {
+			throw new IllegalArgumentException("NULL list of trackpoints passed to constructor.");
+		}
+		
+		this.points = points;
+	}
+	
+	/**
+	 * Add a point to the list of track points.
+	 * 
+	 * @param point A point to add to the track point list, cannot be null.
+	 */
+	public void addPoint(GeoTrackPoint point) {
+		if(points == null) {
+			throw new IllegalArgumentException("Attempted to add NULL value to track point list.");
+		}
+		
+		points.add(point);
+	}
+	
+	/**
+	* Adds a new point with the given attributes.
+	*
+	* @param latitude			Latitude of the trackpoint, required
+	* @param longitude			Longitude of the trackpoint, required
+	* @param altitude			Altitude of the trackpoint, maybe null
+	* @param name				Name of trackpoint, maybe null
+	* @param velocity			Velocity in meters/sec, maybe null
+	* @param distanceFromHP		Trackpoint distance from an established "home
+	*							point", maybe null if not applicable
+	* @param distanceTraveled	Overall distance traveled in meters at the
+	*							time this trackpoint was created, maybe null
+	*							if not applicable
+	* @param timestamp			Trackpoint creation time, maybe null if not
+	*							applicable
+	*/
+	public void addPoint(Double latitude,
+				Double longitude,
+				Double altitude,
+				String name,
+				Double velocity,
+				Double distanceFromHP,
+				Double distanceTraveled,
+				Long timestamp) {
+		points.add(new GeoTrackPoint(
+						latitude, 
+						longitude, 
+						altitude, 
+						name, 
+						velocity, 
+						distanceFromHP, 
+						distanceTraveled, 
+						timestamp));
+	}
+	
+	/**
+	 * Returns an iterator over the points in this GeoTrackPointList.
+	 * 
+	 * @return An iterator over the elements of the list.
+	 */
+	public Iterator<GeoTrackPoint> iterator() {
+		return points.iterator();
+	}
+	
+	/**
+	 * Returns true if this list contains no points.
+	 * 
+	 * @return True if this list contains no points.
+	 */
+	public boolean isEmpty() {
+		return points.isEmpty();
+	}
+	
+	/**
+	 * Returns serializes this object to JSON. The JSON string can than be
+	 * used as the TSK_GEO_TRACKPOINTS attribute of the TSK_GPS_TRACK artifact.
+	 * 
+	 * @return JSON string
+	 */
+	public String serialize() {
```

Comment:
The comment suggests what this can just do by default. Could this be changed to 'public BlackboardAttribute toAttribute()'?